main/inc/lib/tracking.lib.php

```diff
@@ -1898,104 +1944,195 @@ public static function getLpStats(
+                                    WHERE
+                                        exe_exo_id = '.(int) $row_path['path'].' AND
+                                        status <> "incomplete" AND
+                                        exe_user_id='.$user_id.' AND
+                                        orig_lp_id = '.(int) $lp_id.' AND
+                                        orig_lp_item_id = '.(int) $lp_item_id.' AND
+                                        c_id = '.$course_id.' AND
+                                        session_id = '.$session_id.'
+                                    ORDER BY exe_date';
+                                    $res_attempts = Database::query($sql);
+                                    $num_attempts = Database::num_rows($res_attempts);
+                                    if ($num_attempts > 0) {
+                                        $n = 1;
+                                        while ($row_attempts = Database::fetch_array($res_attempts)) {
+                                            $my_score = $row_attempts['exe_result'];
+                                            $my_maxscore = $row_attempts['exe_weighting'];
+                                            $my_exe_id = $row_attempts['exe_id'];
+                                            $mktime_start_date = api_strtotime($row_attempts['start_date'], 'UTC');
+                                            $mktime_exe_date = api_strtotime($row_attempts['exe_date'], 'UTC');
+                                            $time_attemp = ' - ';
+                                            if ($mktime_start_date && $mktime_exe_date) {
+                                                $time_attemp = api_format_time($row_attempts['exe_duration'], 'js');
+                                            }
+                                            if (!$is_allowed_to_edit && $result_disabled_ext_all) {
+                                                $view_score = Display::return_icon(
+                                                    'invisible.png',
+                                                    get_lang(
+                                                        'ResultsHiddenByExerciseSetting'
+                                                    )
                                                 );
                                             } else {
-                                                if ($my_maxscore == 0) {
-                                                    $view_score = $my_score;
-                                                } else {
+                                                // Show only float when need it
+                                                if ($my_score == 0) {
                                                     $view_score = ExerciseLib::show_score(
-                                                        $my_score,
+                                                        0,
```

Comment:
Logic change (order and types of checks). To be kept under surveillance (may affect quiz type, at first glance).